api/src/main/java/com/codeforcommunity/dto/report/ReportWithLibrary.java

```diff
@@ -178,4 +178,80 @@ public String getParentSupport() {
   public void setParentSupport(String parentSupport) {
     this.parentSupport = parentSupport;
   }
+
+  public String toHeaderCSV() {
+    return "Report ID,"
+        + "Created At,"
+        + "Updated At,"
+        + "School ID,"
+        + "User ID,"
+        + "Number of Children,"
+        + "Number of Books,"
+        + "Most Recent Shipment Year,"
+        + "Library Status,"
+        + "Visit Reason,"
+        + "Is Shared Space,"
+        + "Has Inviting Space,"
+        + "Assigned Person Role,"
+        + "Assigned Person Title,"
+        + "Apprenticeship Program,"
+        + "Trains Mentors and Apprentices,"
+        + "Has Checkin Timetables,"
+        + "Has Book Checkout System,"
+        + "Number of Student Librarians,"
+        + "Has Sufficient Training,"
+        + "Teacher Support,"
+        + "Parent Support\n";
+  }
+
+  public String toRowCSV() {
+    return this.getId().toString()
+        + ","
+        + (this.getCreatedAt() == null ? "" : this.getCreatedAt().toString())
+        + ","
+        + (this.getUpdatedAt() == null ? "" : this.getUpdatedAt().toString())
+        + ","
+        + (this.getSchoolId() == null ? "" : this.getSchoolId().toString())
+        + ","
+        + (this.getUserId() == null ? "" : this.getUserId().toString())
+        + ","
+        + (this.getNumberOfChildren() == null ? "" : this.getNumberOfChildren().toString())
+        + ","
```

Comment:
This is incredibly long, I think we should use a for loop here. Logically, all we're saying is: 
```
StringBuilder builder = new StringBuilder();
for (Object field: allFields) {
  if (field != null) builder.append(field.toString())
  field.append(','); // always append comma for each field
}
```
The only question is how to get an iterable list of all fields.